src/Faker/Generator/MersenneTwisterIntegerGenerator.php

```diff
@@ -0,0 +1,42 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Faker\Generator;
+
+/**
+ * @experimental
+ */
+final class MersenneTwisterIntegerGenerator implements SeedableIntegerGenerator
```

Comment:
Naming of this class is probably not ideal. Specifically this is PHP's global instance of Mt19937 (colloquially “Mersenne Twister”). There's also https://www.php.net/manual/en/class.random-engine-mt19937.php which is the OO variant that does not rely on global state.

Reply:
@TimWolla 

Yes, it is.

For PHP 8.3, an implementation that uses `Random\Randomizer` could look like this:


```php
<?php

declare(strict_types=1);

namespace Faker\Core\Generator;

use Random\Randomizer;

/**
 * @experimental
 */
final class RandomIntegerGenerator implements IntegerGenerator
{
    private Randomizer $randomizer;

    public function __construct(Randomizer $randomizer)
    {
        $this->randomizer = $randomizer;
    }

    public function integer(): int
    {
        return $this->randomizer->nextInt();

    }

    public function integerBetween(int $minimum, int $maximum): int
    {
        if ($minimum > $maximum) {
            throw new \InvalidArgumentException(sprintf(
                'Minimum value %d should not be greater than the maximum value %d.',
                $minimum,
                $maximum,
            ));
        }

        return $this->randomizer->getInt($minimum, $maximum);
    }

    public function largestInteger(): int
    {
        return PHP_INT_MAX;
    }
}
```

What do you think?

Reply:
Depending on the expected semantics of the various methods, the implementation is broken. `->nextInt()` is not guaranteed to return values up to `PHP_INT_MAX`, so if you rely on that, you will be disappointed.

Generally I'd say that using `->nextInt()` is an anti-pattern (similarly to calling `mt_rand()` without parameters; the method only exists for drop-in compatibility of the OO API). Retrieving an arbitrary positive integer is *very* rarely useful. You generally know what range you want to retrieve and `->getInt(0, PHP_INT_MAX)` would at least be explicit about the fact that you do not care.

Almost all uses of `mt_rand()` without parameters are better replaced by `random_bytes()` (possibly combined with `bin2hex()`).